.github/workflows/ci.yml

```
@@ -150,3 +150,19 @@ jobs:
         env:
           GOARCH: ${{ matrix.arch }}
         run: make test-short
+
+  test-compatibility:
+    runs-on: ubuntu-latest
+    needs: [compatibility-test]
+    steps:
+      - name: Print result
+        run: echo ${{ needs.compatibility-test.result }}
+      - name: Interpret result
+        run: |
+          if [[ success == ${{ needs.compatibility-test.result }} ]]
+          then
+            echo "All matrix jobs passed!"
+          else
+            echo "One or more matrix jobs failed."
+            false
+          fi
```

nit: Why do we have two steps?

```suggestion
      - name: Test if compatibility-test passed
        run: |
          echo "${{ needs.compatibility-test.result }}"
          test "${{ needs.compatibility-test.result" }} == "success"
```

---

Thanks for the suggestion, quotes were not aligned with the suggestion, so did a local commit and updated the PR.